src/api-docs/swagger.ts

```diff
@@ -0,0 +1,176 @@
+import { version } from 'mongoose';
+import swaggerAutogen from 'swagger-autogen';
+
+const doc = {
+  info: {
+    title: 'OrganizeMe API',
+    version: '1.0.0',
+    description: 'API documentation for the OrganizeMe project',
+  },
+  host: 'localhost:3000',
+  schemes: ["http"],
+  components: {
+    schemas: {
+      User: {
+        type: 'object',
+        properties: {
+          githubId: {
+            type: 'string',
+            description: 'The unique GitHub ID of the user',
+          },
+          username: {
+            type: 'string',
+            description: 'The username of the user',
+          },
+          profileImgUrl: {
+            type: 'string',
+            description: 'URL to the user\'s profile image',
+          },
+          displayName: {
+            type: 'string',
+            description: 'The display name of the user',
+          },
+          canvasToken: {
+            type: 'string',
+            description: 'Token used for authentication with Canvas',
+          },
+        },
+        required: ['githubId', 'username', 'displayName'],
+      },
+      Task: {
+        type: 'object',
+        properties: {
+          _id: {
+            type: 'string',
+            description: 'The unique identifier for the task',
+          },
+          title: {
+            type: 'string',
+            description: 'The title of the task',
+          },
+          description: {
+            type: 'string',
+            description: 'Details about the task',
+          },
+          dueDate: {
+            type: 'string',
+            format: 'date',
+            description: 'The due date of the task',
+          },
+          status: {
+            type: 'string',
+            description: 'The current status of the task',
+            enum: ['pending', 'in-progress', 'completed'],
+          },
```

Comment:
Would it be a good idea to put a "default: pending" or something here? It might make it easier to build the frontend because then we don't have to specify every time.